src/DataKeyAwareTrait.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Dhii\Data\Container;
+
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+
+/**
+ * Functionality for storage and retrieval of a data key.
+ *
+ * @since [*next-version*]
+ */
+trait DataKeyAwareTrait
+{
+    /**
+     * The data key, if any.
+     *
+     * @since [*next-version*]
+     *
+     * @var string|Stringable|null
+     */
+    protected $dataKey;
+
+    /**
+     * Assigns a data key to this instance.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable|null $key The key.
+     *
+     * @return $this
```

Comment:
Missing `throws` tag

Reply:
Actually, my idea is that perhaps it should become a standard that any method that accepts an argument can throw an `InvalidArgumentException`, and we can avoid documenting this.